Source/Particles/ParticleBoundaries_K.H

```diff
@@ -61,16 +66,38 @@ namespace ApplyParticleBoundaries {
                 x = 2*xmax - x;
                 change_sign_ux = true;
             }
+            else if (xmax_bc == ParticleBoundaryType::Thermal) {
+                x = 2*xmax - x;
+                rethermalize_x = true;
+            }
         }
     }
 
-    /* \brief Applies absorbing or reflecting boundary condition to the input particles, along all axis.
+    /* \brief Thermalize particles that have been identified to cross the boundary.
+     *        The normal component samples from a half-Maxwellian,
+     *        and the two tangential components will sample from full Maxwellian disbutions
+     *        with thermal velocity uth
+     */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    void thermalize_boundary_particle (amrex::ParticleReal& u_norm, amrex::ParticleReal& u_tang1,
+                                       amrex::ParticleReal& u_tang2, amrex::Real uth,
+                                       amrex::RandomEngine const& engine)
+    {
+        u_tang1 = (uth > 0._rt) ? PhysConst::c * amrex::RandomNormal(0._rt, uth, engine) : 0._rt;
+        u_tang2 = (uth > 0._rt) ? PhysConst::c * amrex::RandomNormal(0._rt, uth, engine) : 0._rt;
+        u_norm = -1._rt * PhysConst::c * generateGaussianFluxDist(0._rt, uth, engine) * ((u_norm>0)-(u_norm<0));
```

Comment:
This would be the more standard usage
```suggestion
        u_norm = std::copysign(1._prt, -u_norm) * PhysConst::c * generateGaussianFluxDist(0._rt, uth, engine);
```

Reply:
nice! Thanks!